migrate to scaffold-ui

technophile-04 opened the pull request:
### Description: 

- migration to [scaffold-ui](https://github.com/scaffold-eth/scaffold-ui) (#1136)
- Update Alchemy API key to new rolling key for tracking new/old requests (#1190)

rin-st commented:
Oh sh, it will break all the extensions which use components moved to `scaffold-ui`

<img width="901" height="229" alt="image" src="https://github.com/user-attachments/assets/61d6c088-df3f-4844-83c0-d2dd4877fc7f" />

technophile-04 commented:
Created a [codemode CLI](https://github.com/technophile-04/create-eth-codemod/): 

https://github.com/user-attachments/assets/fed47f43-c9dd-4980-be70-027c8ad26e4a

<details>

<summary>
    
## What The Codemod Handles
    
</summary>



- Rewrites component imports (`~~/components/scaffold-eth`, `~~/components/scaffold-eth/Input`, `~~/components/scaffold-eth/Input/AddressInput`, `~~/components/scaffold-eth/Address/Address`, etc.) to `@scaffold-ui/components`.
- Renames the `InputBase` specifier to `BaseInput` (`{ BaseInput as InputBase }`).
- Emits warnings whenever it detects:
  - `Address`, with a link to the updated usage example showing the new `chain` and `blockExplorerAddressLink` props.
  - `EtherInput`, highlighting the new `onValueChange({ valueInEth })` callback pattern with a reference example.
  - `isENS`, reminding you to import it from `~~/utils/scaffold-eth/common`.

</details>



This will help us migrate all the current extensions nicely 🙌 Umm, regarding the third-party extension, maybe we can send the message in TG groups to make use of this CLI? 

cc @carletex once as well 

---- 

Also regarding the [create-eth-codemod repo](https://github.com/technophile-04/create-eth-codemod/tree/master) repo, I didn't create that CLI here in this repo because I think codemod's will be helpful only for extension developers? And don't want to ship the codemod code to people scaffolding create-eth app. We can move that repo under Scaffold-ETH org  as well 🙌 what do you guys say?

rin-st commented:
I've read the code and tested codemod, working great!

> Umm, regarding the third-party extension, maybe we can send the message in TG groups to make use of this CLI?

Yep, I think it's enough

> Also regarding the [create-eth-codemod repo](https://github.com/technophile-04/create-eth-codemod/tree/master) repo, I didn't create that CLI here in this repo because I think codemod's will be helpful only for extension developers? And don't want to ship the codemod code to people scaffolding create-eth app. We can move that repo under Scaffold-ETH org as well 🙌 what do you guys say?

Agree, let's just move it under Scaffold-eth

---

So, we migrate curated extensions first and then merge this PR when gh actions succeed, right?

technophile-04 commented:
> So, we migrate curated extensions first and then merge this PR when gh actions succeed, right? 

I just created the [migration PR's to create-eth-extension](https://github.com/scaffold-eth/create-eth-extensions/pulls?q=sort:updated-desc+is:pr+is:open) repo, also tested them all with gh-actions <3 checkout this [results](https://github.com/scaffold-eth/create-eth/actions/runs/19061076170?pr=355) . Also merged `example` branch PR in that repo so we get passing actions here. 

We can merge this PR now and then merge all the extension's PR in create-eth-extension repo once we launch a new version 🙌 


Also at [`8522371` (#355)](https://github.com/scaffold-eth/create-eth/pull/355/commits/8522371fbf18f4f1200b8adc2a49f9e414f31b24) I changed the version from minor to major bump. So new version will be 2.x.x 

Umm not sure if major bump is overkill since no major changes happed to CLI itself, but maybe good enough justified since it requires the extensions to updated as well as lot of changes to base templates files?

rin-st commented:
> I just created the [migration PR's to create-eth-extension](https://github.com/scaffold-eth/create-eth-extensions/pulls?q=sort:updated-desc+is:pr+is:open) repo, also tested them all with gh-actions <3 checkout this [results](https://github.com/scaffold-eth/create-eth/actions/runs/19061076170?pr=355) . Also merged example branch PR in that repo so we get passing actions here.

Great job Shiv, thank you!
Have you checked other curated extensions: eip-712, randao, porto, envio, drizzle-neon? No changes for them?

> Also at [8522371 (#355)](https://github.com/scaffold-eth/create-eth/pull/355/commits/8522371fbf18f4f1200b8adc2a49f9e414f31b24) I changed the version from minor to major bump. So new version will be 2.x.x
> Umm not sure if major bump is overkill since no major changes happed to CLI itself, but maybe good enough justified since it requires the extensions to updated as well as lot of changes to base templates files?

Agree, 2.x.x feels right here

---

Todo next:
migrate sre challenges
